### PR TITLE
Add workaround about using Stale Read with TiFlash

### PR DESCRIPTION
**Pull request by leiysky**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add workaround about using Stale Read with TiFlash.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6541
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by leiysky** [comment]
Rendered: https://github.com/pingcap/docs/blob/9c4185891c31133e998dfba8d17250d77c954992/as-of-timestamp.md

**Review by Yisaer — APPROVED** [review]

**Review by ilovesoup — APPROVED** [review]
lgtm

**Comment by leiysky** [comment]
@TomShawn PTAL

File: as-of-timestamp.md [hunk 1]
```diff
@@ -10,6 +10,13 @@ This document describes how to perform the [Stale Read](/stale-read.md) feature
 > **Warning:**
 >
 > Currently, Stale Read is an experimental feature. It is not recommended to use it in the production environment.
+>
+> Stale Read can not work with TiFlash now. If your SQL query contains `AS OF TIMESTAMP` clause and TiDB would possibly read from TiFlash replica, then you might encounter error  with message like `ERROR 1105 (HY000): stale requests require tikv backend`.
+>
+> You can fix the problem by disabling TiFlash replica for your Stale Read query as follows:
+> 
+> - Use variable to let TiDB read from TiKV `set session tidb_isolation_read_engines='tidb,tikv'`.
+> - Use [hint](https://docs.pingcap.com/tidb/dev/optimizer-hints#read_from_storagetiflasht1_name--tl_name--tikvt2_name--tl_name-) to enforce TiDB to read from TiKV.
```
Comment (TomShawn) [inline as-of-timestamp.md:19]:
```suggestion
> - Use the `set session tidb_isolation_read_engines='tidb,tikv'` variable.
> - Use the [hint](/optimizer-hints.md#read_from_storagetiflasht1_name--tl_name--tikvt2_name--tl_name-) to enforce TiDB to read data from TiKV.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**